lottie/src/main/java/com/airbnb/lottie/network/NetworkFetcher.java

```diff
@@ -80,69 +68,41 @@ private LottieComposition fetchFromCache() {
     return null;
   }
 
+  @NonNull
   @WorkerThread
-  private LottieResult<LottieComposition> fetchFromNetwork() {
-    try {
-      return fetchFromNetworkInternal();
-    } catch (IOException e) {
-      return new LottieResult<>(e);
-    }
-  }
-
-  @WorkerThread
-  private LottieResult<LottieComposition> fetchFromNetworkInternal() throws IOException {
+  private LottieResult<LottieComposition> fetchFromNetwork(@NonNull String url, @Nullable String cacheKey) {
     Logger.debug("Fetching " + url);
 
-
-    HttpURLConnection connection = (HttpURLConnection) new URL(url).openConnection();
-    connection.setRequestMethod("GET");
-
+    LottieFetchResult fetchResult = null;
     try {
-      connection.connect();
-
-      if (connection.getErrorStream() != null || connection.getResponseCode() != HttpURLConnection.HTTP_OK) {
-        String error = getErrorFromConnection(connection);
-        return new LottieResult<LottieComposition>(new IllegalArgumentException("Unable to fetch " + url + ". Failed with " + connection.getResponseCode() + "\n" + error));
+      fetchResult = fetcher.fetchSync(url);
+      if (fetchResult.isSuccessful()) {
+        InputStream inputStream = fetchResult.bodyByteStream();
+        String contentType = fetchResult.contentType();
+        LottieResult<LottieComposition> result = fromInputStream(url, inputStream, contentType, cacheKey);
+        Logger.debug("Completed fetch from network. Success: " + (result.getValue() != null));
+        return result;
+      } else {
+        return new LottieResult<>(new IllegalArgumentException(fetchResult.error()));
       }
-
-      LottieResult<LottieComposition> result = getResultFromConnection(connection);
-      Logger.debug("Completed fetch from network. Success: " + (result.getValue() != null));
-      return result;
     } catch (Exception e) {
       return new LottieResult<>(e);
     } finally {
-      connection.disconnect();
-    }
-  }
-
-  private String getErrorFromConnection(HttpURLConnection connection) throws IOException {
-    int responseCode = connection.getResponseCode();
-    BufferedReader r = new BufferedReader(new InputStreamReader(connection.getErrorStream()));
-    StringBuilder error = new StringBuilder();
-    String line;
-
-    try {
-      while ((line = r.readLine()) != null) {
-        error.append(line).append('\n');
-      }
-    } catch (Exception e) {
-      throw e;
-    } finally {
-      try {
-        r.close();
-      } catch (Exception e) {
-        // Do nothing.
+      if (fetchResult != null) {
+        try {
+          fetchResult.close();
+        } catch (IOException e) {
+          e.printStackTrace();
```

Comment:
`e.printStackTrace();` should not be called in Android. You can use `Logger.warning`

Reply:
done